### PR TITLE
feat(settings): Add new settings screen

### PR DESCRIPTION
**Pull request by damassi**
### Description

This PR refactors / updates the settings screen, and makes the offline sync experience a bit nicer with some animated ellipses. 

https://user-images.githubusercontent.com/236943/209248440-3f5dc584-9b8c-4f02-aa0c-489c5cb22f5c.mp4

cc @artsy/mobile-platform 

### PR Checklist

- [ ] I tested my changes on **iOS** and **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -65,6 +65,7 @@
     "react-dom": "^17.0.2",
     "react-nanny": "^2.15.0",
     "react-native": "0.66.2",
+    "react-native-animated-ellipsis": "^2.0.0",
```
Comment (damassi) [inline package.json:68]: @admbtlr - i tried using yours, but no files in `dist` and didn't want to run a precompile step! This works good tho.

File: src/typings/reactNativeElipses.d.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    dots: string[]
+    animation: Animated.CompositeAnimation
+    constructor(props: AnimatedEllipsisProps)
+    componentDidMount(): void
+    componentWillUnmount(): void
+    startAnimation(): void
+    stopAnimation(): void
+    render(): JSX.Element
+  }
+}
```
Comment (damassi) [inline src/typings/reactNativeElipses.d.ts:23]: I hit tab on copilot

File: src/app/screens/Settings/PresentationModeSettings.tsx [hunk 3]
```diff
@@ -1,9 +1,14 @@
 import { Flex, Spacer, Text } from "@artsy/palette-mobile"
+import { FullScreenDivider } from "app/components/FullScreenDivider"
```
Comment (pvinis) [inline src/app/screens/Settings/PresentationModeSettings.tsx:2]: what's this for?

File: src/app/components/FullScreenDivider.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { FlexProps, Separator } from "@artsy/palette-mobile"
+import { Screen } from "palette"
+
+export const FullScreenDivider: React.FC<FlexProps> = (flexProps) => {
```
Comment (pvinis) [inline src/app/components/FullScreenDivider.tsx:4]: should this be under Screen? 🤔 like Screen.FullWidthDivider

File: src/app/screens/Settings/Settings.tsx [hunk 5]
```diff
@@ -6,185 +6,108 @@ import {
...
+          </Touchable>
+
+          <Button block onPress={() => void GlobalStore.actions.auth.signOut()}>
+            Log out
+          </Button>
+        </Join>
+
+        <Separator my={4} />
+
+        {__DEV__ && (
```
Comment (pvinis) [inline src/app/screens/Settings/Settings.tsx:81]: 👌👌

File: src/typings/reactNativeElipses.d.ts (hunk 2 above)
Comment (pvinis) [reply src/typings/reactNativeElipses.d.ts:23]: is this lib just not using TS at all?

**Review by pvinis — APPROVED** [review]
a couple of tiny things, but non are blocking. see what you think and merge when ready.

File: src/typings/reactNativeElipses.d.ts (hunk 2 above)
Comment (damassi) [reply src/typings/reactNativeElipses.d.ts:23]: Nope, doesn't seem like it, just .js

File: src/app/components/FullScreenDivider.tsx (hunk 4 above)
Comment (damassi) [reply src/app/components/FullScreenDivider.tsx:4]: Oh yeah, i can do that!

File: src/app/screens/Settings/PresentationModeSettings.tsx (hunk 3 above)
Comment (damassi) [reply src/app/screens/Settings/PresentationModeSettings.tsx:2]: Its like a hz line, but stretched and overflowing out of the page bounds. Seems super useful (will move it per comment below)

**Pull request merged**